File: API/models/detalle_pedido.py
from pydantic import BaseModel, Field


class DetallePedidoBase(BaseModel):
    pedido_id: int
    autoparte_id: int
    cantidad: int
    precio_historico: float


class DetallePedidoLineaCreate(BaseModel):
    autoparte_id: int
    cantidad: int = Field(..., gt=0)
    precio_historico: float = Field(..., ge=0)

File: Laravel/app/Models/Macuin/DetallePedido.php
<?php

namespace App\Models\Macuin;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class DetallePedido extends Model
{
    protected $table = 'detalles_pedidos';

    public $timestamps = false;

    protected $casts = [
        'cantidad' => 'integer',
        'precio_historico' => 'decimal:2',
    ];

    protected $fillable = [
        'pedido_id',
        'autoparte_id',
        'cantidad',
        'precio_historico',
    ];

    public function pedido(): BelongsTo
    {
        return $this->belongsTo(Pedido::class, 'pedido_id');
    }

    public function autoparte(): BelongsTo
    {
        return $this->belongsTo(Autoparte::class, 'autoparte_id');
    }
}
